### PR TITLE
Don't treat 2xx status codes as error

### PR DESCRIPTION
**Pull request by duricanikolic**
#### What this PR does
In https://github.com/grafana/mimir/pull/5979 status codes 2xx were, erroneously, treated as error codes. Although unrealistic, this cases should be removed.

Moreover, in https://github.com/grafana/mimir/pull/5584 some entries in CHANGELOG were, erroneously, duplicated. They are removed by this PR.

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by pracucci — APPROVED** [review]
Thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,11 +12,9 @@
 * [ENHANCEMENT] Expose `/sync/mutex/wait/total:seconds` Go runtime metric as `go_sync_mutex_wait_total_seconds_total` from all components. #5879
 * [ENHANCEMENT] Query-scheduler: improve latency with many concurrent queriers. #5880
 * [ENHANCEMENT] Implement support for `limit`, `limit_per_metric` and `metric` parameters for `<Prometheus HTTP prefix>/api/v1/metadata` endpoint. #5890
-* [ENHANCEMENT] Go: updated to 1.21.1. #5955
```
Comment (pstibrany) [inline CHANGELOG.md:15]: Why was this removed?

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,11 +12,9 @@
...
 * [ENHANCEMENT] Distributor: add experimental support for storing metadata when ingesting metrics via OTLP. This makes metrics description and type available when ingesting metrics via OTLP. Enable with `-distributor.enable-otlp-metadata-storage=true`. #5693
 * [ENHANCEMENT] Ingester: added support for sampling errors, which can be enabled by setting `-ingester.error-sample-rate`. This way each error will be logged once in the configured number of times. All the discarded samples will still be tracked by the `cortex_discarded_samples_total` metric. #5584 #6014
 * [ENHANCEMENT] Ruler: Fetch secrets used to configure TLS on the Alertmanager client from Vault when `-vault.enabled` is true. #5239
-* [BUGFIX] Ingester: fix spurious `not found` errors on label values API during head compaction. #5957
```
Comment (pstibrany) [inline CHANGELOG.md:19]: Why was this removed?

File: CHANGELOG.md (hunk 1 above)
Comment (duricanikolic) [reply CHANGELOG.md:15]: @pstibrany It was a duplicate: it is already present in CHANGELOG [here](https://github.com/grafana/mimir/blob/main/CHANGELOG.md?plain=1#L152). The reason: between creation of this PR and its merging, another release of mimir has happened.

File: CHANGELOG.md (hunk 2 above)
Comment (duricanikolic) [reply CHANGELOG.md:19]: @pstibrany It was a duplicate: it is already present in CHANGELOG [here](https://github.com/grafana/mimir/blob/main/CHANGELOG.md?plain=1#L167). The reason: between creation of this PR and its merging, another release of mimir has happened.
